alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -0,0 +1,255 @@
+                            deleteMonitors(monitorIdsToBeDeleted, RefreshPolicy.IMMEDIATE)
+                        }
+                    }
+                    actionListener.onResponse(DeleteWorkflowResponse(deleteResponse.id, deleteResponse.version))
+                } else {
+                    actionListener.onFailure(
+                        AlertingException(
+                            "Not allowed to delete this workflow!",
+                            RestStatus.FORBIDDEN,
+                            IllegalStateException()
+                        )
+                    )
+                }
+            } catch (t: Exception) {
+                if (t is IndexNotFoundException) {
+                    actionListener.onFailure(
+                        OpenSearchStatusException(
+                            "Workflow not found.",
+                            RestStatus.NOT_FOUND
+                        )
+                    )
+                } else {
+                    actionListener.onFailure(AlertingException.wrap(t))
+                }
+            }
+        }
+
+        private suspend fun deleteMonitors(monitorIds: List<String>, refreshPolicy: RefreshPolicy) {
+            if (monitorIds.isEmpty())
+                return
+
+            for (monitorId in monitorIds) {
+                val deleteRequest = DeleteMonitorRequest(monitorId, refreshPolicy)
+                val searchResponse: DeleteMonitorResponse = client.suspendUntil {
+                    AlertingPluginInterface.deleteMonitor(this as NodeClient, deleteRequest, it)
+                }
+            }
+        }
+
+        /**
+         * Returns lit of monitor ids belonging only to a given workflow
+         * @param workflowIdToBeDeleted Id of the workflow that should be deleted
+         * @param monitorIds List of delegate monitor ids (underlying monitor ids)
+         */
+        private suspend fun getDeletableDelegates(workflowIdToBeDeleted: String, monitorIds: List<String>, user: User?): List<String> {
+            // Retrieve monitors belonging to another workflows
+            val queryBuilder = QueryBuilders.boolQuery().mustNot(QueryBuilders.termQuery("_id", workflowIdToBeDeleted)).filter(
+                QueryBuilders.nestedQuery(
+                    Workflow.WORKFLOW_DELEGATE_PATH,
+                    QueryBuilders.boolQuery().must(
+                        QueryBuilders.termsQuery(
+                            Workflow.WORKFLOW_MONITOR_PATH,
+                            monitorIds
+                        )
+                    ),
+                    ScoreMode.None
+                )
+            )
+
+            val searchRequest = SearchRequest()
+                .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
+                .source(SearchSourceBuilder().query(queryBuilder))
+
+            // Check  if user can access the monitors (since the monitors could get modified later and the user might not have the backend roles to access the monitors)
```

Comment:
Nitpick: remove extra space (` `) between `Check` and `if`

Reply:
Good catch!